packages/backend/src/core/discovery/utils/hashDatabase.ts

```diff
@@ -0,0 +1,31 @@
+import { ChainId, ConfigReader } from '@l2beat/discovery'
+import { createHash } from 'crypto'
+import { readFileSync } from 'fs'
+
+export async function getDiscoveryHash(projectName: string, chainId: ChainId) {
+  const configReader = new ConfigReader()
+  const curDiscovery = await configReader.readDiscovery(projectName, chainId)
+  const hasher = createHash('sha1')
+  hasher.update(JSON.stringify(curDiscovery))
+  return `0x${hasher.digest('hex')}`
+}
+
+export function getHashesDatabase(path: string): Record<string, string> {
+  let result = undefined
+
+  try {
+    const hashContent = readFileSync(path, 'utf-8')
+    result = JSON.parse(hashContent) as Record<string, string>
+  } catch (_) {
+    result = {}
```

Comment:
I suggest removing the whole try/catch block. Why do we want to silently fail?

Reply:
To be honest, it's one of those _I'll do something clever in here_. The idea behind it is that if the file does not exists - `readFileSync` throws - or the content of the file is somehow corrupt and parsing it as JSON fails - also throws - just start with an empty "database". I don't really mind this being like this, lmk what you think

Reply:
From the point of view of the code reviewer it's unclear (or seemingly incorrect) of what happens here and why. It either needs an extensive comment, or simply to implement the intention directly - rather than try/catch, simply check if file exists and return {} if not. I prefer the "direct implementation" approach.

As for other unexpected errors during file reading, I see no reason of silently suppressing them. I prefer to fail early - this way you don't let any potential error to propagate further and pollute the rest of the system.